src/test/java/org/folio/rest/unit/ItemDamagedStatusAPIUnitTest.java

```diff
@@ -0,0 +1,333 @@
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.postItemDamagedStatuses(
+      DEFAULT_LANGUAGE,
+      new ItemDamageStatus(),
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(500, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void getItemDamagedStatusesByIdShouldReturnInternalServerError(TestContext testContext) {
+    doThrow(new RuntimeException(INTERNAL_SERVER_ERROR_MSG))
+      .when(postgresClient)
+      .getById(
+        eq(REFERENCE_TABLE),
+        anyString(),
+        any(Class.class),
+        any(Future.class)
+      );
+
+    Async async = testContext.async();
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.getItemDamagedStatusesById(
+      UUID.randomUUID().toString(),
+      DEFAULT_LANGUAGE,
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(500, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void deleteItemDamagedStatusesByIdShouldReturnInternalServerError(TestContext testContext) {
+    doThrow(new RuntimeException(INTERNAL_SERVER_ERROR_MSG))
+      .when(postgresClient)
+      .delete(
+        eq(REFERENCE_TABLE),
+        anyString(),
+        any(Future.class)
+      );
+
+    Async async = testContext.async();
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.deleteItemDamagedStatusesById(
+      UUID.randomUUID().toString(),
+      DEFAULT_LANGUAGE,
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(500, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void deleteItemDamagedStatusesByIdShouldReturnBadRequest(TestContext testContext) {
+    GenericDatabaseException exception = new GenericDatabaseException(new ErrorMessage(getUUIDErrorMap()));
+    doAnswer(setExceptionForHandlerArgument(2, exception))
+      .when(postgresClient)
+      .delete(
+        eq(REFERENCE_TABLE),
+        anyString(),
+        any(Future.class)
+      );
+
+    Async async = testContext.async();
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.deleteItemDamagedStatusesById(
+      UUID.randomUUID().toString(),
+      DEFAULT_LANGUAGE,
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(400, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void putItemDamagedStatusesByIdShouldReturnBadRequest(TestContext testContext) {
```

Comment:
When would a PUT respond with a bad request?

Reply:
wrong UUID

Reply:
By wrong do you mean a UUID of a record that does not already exist? Or an invalid UUID?

Reply:
I think it doesn't matter